libsqlite3-sys/build.rs

```diff
@@ -66,23 +66,39 @@ mod build_bundled {
             .flag("-DSQLITE_DEFAULT_FOREIGN_KEYS=1")
             .flag("-DSQLITE_ENABLE_API_ARMOR")
             .flag("-DSQLITE_ENABLE_COLUMN_METADATA")
-            .flag("-DSQLITE_ENABLE_DBSTAT_VTAB")
-            .flag("-DSQLITE_ENABLE_FTS3")
-            .flag("-DSQLITE_ENABLE_FTS3_PARENTHESIS")
-            .flag("-DSQLITE_ENABLE_FTS5")
             .flag("-DSQLITE_ENABLE_JSON1")
-            .flag("-DSQLITE_ENABLE_LOAD_EXTENSION=1")
-            .flag("-DSQLITE_ENABLE_MEMORY_MANAGEMENT")
-            .flag("-DSQLITE_ENABLE_RTREE")
             .flag("-DSQLITE_ENABLE_STAT2")
             .flag("-DSQLITE_ENABLE_STAT4")
             .flag("-DSQLITE_SOUNDEX")
             .flag("-DSQLITE_THREADSAFE=1")
-            .flag("-DSQLITE_USE_URI")
             .flag("-DHAVE_USLEEP=1")
             .flag("-D_POSIX_THREAD_SAFE_FUNCTIONS") // cross compile with MinGW
             .warnings(false);
 
+        if cfg!(feature = "os-other") {
+            cfg.flag("-DSQLITE_OS_OTHER")
```

Comment:
other than this line and the `-D_FORTIFY_SOURCE`, eliding the optional features is intended to keep the build small and secure on embedded-like platforms. I can revert those changes, if desired.

Reply:
Hmm, I guess this would be a reason not to have it auto-detected based on the value of `TARGET`...

Reply:
I think that might work, actually. I'm using this flag to compile to `x86_64-fortanix-unknown-sgx` where mutexes and filesystems don't quite work as they do on unix or windows. Maybe `if cfg!(target_env = "sgx")` is correct–just not as general.